src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -20,21 +21,21 @@ namespace Microsoft.VisualStudio.TestPlatform.Client
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
     using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers;
     using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
-    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Host;
-    using Microsoft.VisualStudio.TestPlatform.Common.Hosting;
 
     /// <summary>
     /// Implementation for TestPlatform
     /// </summary>
     public class TestPlatform : ITestPlatform
     {
-        private IFileHelper fileHelper;
+        private readonly TestRuntimeProviderManager testHostProviderManager;
+
+        private readonly IFileHelper fileHelper;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="TestPlatform"/> class.
         /// </summary>
-        public TestPlatform() : this(new TestEngine(), new FileHelper())
+        public TestPlatform() : this(new TestEngine(), new FileHelper(), TestRuntimeProviderManager.Instance)
```

Comment:
I'm super confused: Whose responsibility is the TestRuntimeProvider? Not the engine's anymore? How would this look if another engine came in instead of the CrossPlatEngine? Is that even interesting anymore? We might need a doc for this please.